django/test/utils.py

```diff
@@ -111,7 +111,10 @@ def setup_test_environment():
     settings.EMAIL_BACKEND = 'django.core.mail.backends.locmem.EmailBackend'
 
     request._original_allowed_hosts = settings.ALLOWED_HOSTS
-    settings.ALLOWED_HOSTS = ['*']
+    # In addition to hosts defined in settings, support the one default host
+    # for unit tests here. If others are needed, they must be manually added
+    # for the relevant test with override_settings().
+    settings.ALLOWED_HOSTS += ['testserver']
```

Comment:
Good catch! Thanks.
On Jun 3, 2016 9:07 PM, "Simon Charette" <notifications@github.com> wrote:

> In django/test/utils.py
> <https://github.com/django/django/pull/6713#discussion_r65796892>:
>
> > @@ -111,7 +111,10 @@ def setup_test_environment():
> >      settings.EMAIL_BACKEND = 'django.core.mail.backends.locmem.EmailBackend'
> >
> >      request._original_allowed_hosts = settings.ALLOWED_HOSTS
> > -    settings.ALLOWED_HOSTS = ['*']
> > +    # In addition to hosts defined in settings, support the one default host
> > +    # for unit tests here. If others are needed, they must be manually added
> > +    # for the relevant test with override_settings().
> > +    settings.ALLOWED_HOSTS += ['testserver']
>
> Use settings.ALLOWED_HOSTS = settings.ALLOWED_HOSTS + ['testserver'] to
> avoid altering the original list assigned to
> request._original_allowed_hosts above.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/django/django/pull/6713/files/0c6bae16ca933492e5c016c4c28d42f876842af1#r65796892>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAJxhConX5d_pJTnQ65YNoghE2PEElmVks5qIPnrgaJpZM4It8i4>
> .
>